units/commit_queue.go

```diff
@@ -432,53 +410,66 @@ func (j *commitQueueJob) processGitHubPRItem(ctx context.Context, cq *commitqueu
 		return
 	}
 	for _, modulePR := range modulePRs {
-		j.AddError(thirdparty.SendCommitQueueGithubStatus(ctx, j.env, modulePR, message.GithubStatePending, "preparing to test merge", patchDoc.Id.Hex()))
+		j.AddError(thirdparty.SendCommitQueueGithubStatus(ctx, j.env, modulePR, message.GithubStatePending, "preparing to test merge", v.Id))
 	}
 
 	event.LogCommitQueueStartTestEvent(v.Id)
 }
 
 func (j *commitQueueJob) processCLIPatchItem(ctx context.Context, cq *commitqueue.CommitQueue, nextItem *commitqueue.CommitQueueItem, projectRef *model.ProjectRef, githubToken string) {
-	patchDoc, err := patch.FindOneId(nextItem.Issue)
+	v, _, err := updateAndFinalizeCqPatch(ctx, j.env.Settings(), nextItem.Issue, commitqueue.SourceDiff, projectRef, false, githubToken)
 	if err != nil {
-		err = errors.Wrapf(err, "finding patch '%s'", nextItem.Issue)
 		j.logError(err, *nextItem)
 		event.LogCommitQueueEnqueueFailed(nextItem.Issue, err)
 		j.dequeue(ctx, cq, *nextItem, err.Error())
 		return
 	}
-	if patchDoc == nil {
-		err := errors.Errorf("patch '%s' not found", nextItem.Issue)
+
+	nextItem.Version = v.Id
+	nextItem.ProcessingStartTime = time.Now()
+	if err = cq.UpdateVersion(nextItem); err != nil {
+		err = errors.Wrap(err, "updating commit queue item's version")
 		j.logError(err, *nextItem)
-		event.LogCommitQueueEnqueueFailed(nextItem.Issue, err)
 		j.dequeue(ctx, cq, *nextItem, err.Error())
 		return
 	}
 
-	project, pp, err := updatePatch(ctx, j.env.Settings(), githubToken, projectRef, patchDoc)
-	if err != nil {
-		err = errors.Wrap(err, "updating patch")
+	if err = setDefaultNotification(v.Author); err != nil {
+		err = errors.Wrap(err, "setting up default commit queue notification")
 		j.logError(err, *nextItem)
-		event.LogCommitQueueEnqueueFailed(nextItem.Issue, err)
-		j.dequeue(ctx, cq, *nextItem, err.Error())
-		return
 	}
+	event.LogCommitQueueStartTestEvent(v.Id)
+}
 
-	pp, err = AddMergeTaskAndVariant(ctx, patchDoc, project, projectRef, commitqueue.SourceDiff)
+func updateAndFinalizeCqPatch(ctx context.Context, settings *evergreen.Settings, patchId, source string, projectRef *model.ProjectRef, returnProjectConfig bool, githubToken string) (*model.Version, *model.Project, error) {
+	patchDoc, err := patch.FindOneId(patchId)
 	if err != nil {
-		err = errors.Wrap(err, "updating patch project config to include merge task")
-		j.logError(err, *nextItem)
-		event.LogCommitQueueEnqueueFailed(nextItem.Issue, err)
-		j.dequeue(ctx, cq, *nextItem, err.Error())
-		return
+		return nil, nil, errors.Wrapf(err, "finding patch '%s'", patchId)
+	}
+	if patchDoc == nil {
+		return nil, nil, errors.Errorf("patch '%s' not found", patchId)
+	}
+
+	var projectConfig *model.Project
+	if returnProjectConfig {
+		projectConfig, _, err = model.GetPatchedProject(ctx, settings, patchDoc, githubToken)
```

Comment:
I think we're using projectConfig wrong here (I recognize this was misnamed from the other file though). GetPatchedProject returns project and projectConfig (and projectConfig is yaml project ref settings, not the translated parser project) so we should rename this to just project. 
i.e. we really have `proj, projectConfig, err = model.GetPatchedProject(...` 

Also, we should just return the one from updatePatch rather than re-calling this here. Then we can remove the return boolean.

Reply:
thanks! done!